lib/sort_by_length.rb

```diff
@@ -1,7 +1,23 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(n^2)
 def sort_by_length(my_sentence)
```

Comment:
This does work, but it's O(n^3) not O(n^2) time complexity and because you're making a pair of arrays, the space complexity is O(n), not O(n^2).  For example if you double the number of words, you're not increasing the memory usage by 4.

I very much encourage you to look at my example code on the sorting algorithms, specifically how we swap items and thus avoid having to use the `ordered_words` array.  Also look below at my comments on the `.insert` method.